kolibri/logger/models.py

```diff
@@ -128,9 +132,23 @@ class UserSessionLog(BaseLogModel):
     user = models.ForeignKey(FacilityUser)
     channels = models.TextField(blank=True)
     start_timestamp = models.DateTimeField(auto_now_add=True)
-    completion_timestamp = models.DateTimeField(blank=True, null=True)
+    last_interaction_timestamp = models.DateTimeField(auto_now=True, null=True)
     pages = models.TextField(blank=True)
 
+    @classmethod
+    def update_log(cls, user):
+        try:
+            user_session_log = cls.objects.filter(user=user).latest('last_interaction_timestamp')
+        except ObjectDoesNotExist:
+            user_session_log = None
+
+        if not user_session_log or timezone.now() - user_session_log.last_interaction_timestamp > timedelta(minutes=5):
+            user_session_log = cls(user=user)
+            user_session_log.save()
+        else:
+            user_session_log.save()
```

Comment:
We might want to do another check here to make sure we're not hammering the database too much (only save if the time is more than a minute more than it was before?)